src/WebJobs.Extensions.DurableTask/ContextInterfaces/IDurableOrchestrationClient.cs

```diff
@@ -191,6 +191,15 @@ public interface IDurableOrchestrationClient
         [Obsolete("This feature is in preview.")]
         Task RewindAsync(string instanceId, string reason);
 
+        /// <summary>
+        /// Gets the status of the specified orchestration instance.
+        /// </summary>
+        /// <param name="instanceId">The ID of the orchestration instance to query.</param>
+        /// <param name="showHistory">Boolean marker for including execution history in the response.</param>
+        /// <param name="showHistoryOutput">Boolean marker for including input and output in the execution history response.</param>
+        /// <returns>Returns a task which completes when the status has been fetched.</returns>
+        Task<DurableOrchestrationStatus> GetStatusAsync(string instanceId, bool showHistory, bool showHistoryOutput);
```

Comment:
I know I've been complaining about extension methods, but if we're using extension methods anyways I wonder if this is a good candidate. I would feel bad for anyone implementing an interface that has multiple overloads for the same method.

Reply:
Unfortunately that doesn't work since extension methods can only access the methods on `IDurableOrchestrationClient`, and this was meant to get around the fact that `Task<DurableOrchestrationStatus> IDurableOrchestrationClient.GetStatusAsync(string instanceId, bool showHistory, bool showHistoryOutput, bool showInput);`  could possibly through a `NotImplementedException`, so we can't use that method in our extension method.

I'm thinking that it would just be better to go back to just silently eat the `showInput` parameter for the small % of customers that use a custom `DurabilityProvider` implementation that does not support the `showInput` parameter.